fix Telekinesis teleporting some items

AlinaStarkova opened the pull request:
Fixed telekinesis from teleporting the Blackbox, bodybags and all its contents, and brains from MMIs - Nari

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

as outlined in issue #55885, telekinesis was teleporting stuff like the blackbox, brains from an MMI, bodybags (including all the contents if bluespace) when activated with a telekinetic hand

fixed by simply having them drop onto the tile its on when using telekinesis, rather than enter the users hand

(the conflicts is caused by a broken PR that got merged, its not to do with my code so e)
https://user-images.githubusercontent.com/40489693/103470740-f85e2880-4d43-11eb-8f19-112a09b4d0a2.mp4


<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

as neato as it is for a way to teleport your objective right to you, among other things, this is a bug with the way TK functions, and TK is just not feasible to overhaul for this

fixes #55885
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl: Nari Harimoto
fix: Telekinesis no longer teleports the Blackbox, brains from an MMI, or bodybags with all its contents, directly to you
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

AlinaStarkova committed:
fix Telekinesis teleporting items

Fixed telekinesis from teleporting the Blackbox, bodybags and all its contents, and brains from MMIs - Nari

LemonInTheDark commented:
Not all items right? just some of em

AlinaStarkova commented:
> 
> 
> Not all items right? just some of em

its just those 3, e, it would be a right pain to check every single item in the game to make them not put items into your hand when activated

Jared-Fogle commented on code/game/machinery/telecomms/machines/message_server.dm:
```diff
@@ -24,7 +24,7 @@
 	. = ..()
 	if(stored)
-		user.put_in_hands(stored)
+		stored.forceMove(get_turf(src))
```
Use `drop_location()` instead, here and everywhere else.

AlinaStarkova replied:
i tried, i really did, i tried drop_location(stored) as well as stored.drop_location() .... all those did, was *delete* the items they were supposed to drop, like, completely vanish. force moving it however worked flawlessly

MrDoomBringer replied:
I believe @Jared-Fogle meant `stored.forceMove(drop_location())`
If you're ever unsure of how to use a particular function or whatever, one way to quickly figure out is to search the codebase for example use cases! :)
![image](https://user-images.githubusercontent.com/29008542/103471426-3b24fe00-4d4e-11eb-9cdf-8bc6202c31f5.png)

If you search for the proc definition (or ctrl+click a function call if you're using vscode and the .dm extensions) you can sometimes see helpful comments explaining the function further.
![image](https://user-images.githubusercontent.com/29008542/103471462-c3a39e80-4d4e-11eb-910b-0d17b5ca6f3e.png)

AlinaStarkova replied:
this is outdated, its already added in.

Jared-Fogle commented:
>  (does make it less convenient, however this was the fix utilized for stopping AIs from ejecting beakers from machines spawning them onto the AI core)

I think you should just check if they're adjacent after dropping it, and put it into their hands then.

AlinaStarkova commented:
> 
> 
> > (does make it less convenient, however this was the fix utilized for stopping AIs from ejecting beakers from machines spawning them onto the AI core)
> 
> I think you should just check if they're adjacent after dropping it, and put it into their hands then.

hmmm, let me go code browsing to find out how to do this

Jared-Fogle commented:
```dm
if (thing_being_dropped.Adjacent(user))
    user.put_in_hands(thing_being_dropped)
```

AlinaStarkova committed:
fixes

fixes new bug where opening the bodybag in hand will make it vanish - Nari

made a better method for pulling items out of the recorder and MMI, if activated by hand it puts item into your hands, if activated with telekinesis, it drops it on its tile - Nari

Jared-Fogle commented:
Please merge from latest upstream, there was a CI failure on master that has since been fixed.

AlinaStarkova committed:
Merge branch 'master' into fix-tk-bodybags-blackbox
needed

Jared-Fogle commented on code/game/objects/items/bodybag.dm:
```diff
@@ -8,7 +8,10 @@
+	if(user.is_holding(src))
+		deploy_bodybag(user, user.loc)
+	else
+		deploy_bodybag(user, src.loc)
```
```suggestion
		deploy_bodybag(user, get_turf(src))
```

AlinaStarkova replied:
done

Jared-Fogle commented on code/game/objects/items/bodybag.dm:
```diff
@@ -8,7 +8,10 @@
 /obj/item/bodybag/attack_self(mob/user)
-	deploy_bodybag(user, user.loc)
+	if(user.is_holding(src))
+		deploy_bodybag(user, user.loc)
```
```suggestion
		deploy_bodybag(user, get_turf(user))
```

ArcaneMusic commented:
If you change the wording on your PR body to something along the lines of "Fixes #issuenumber" it will close the issue once merged, btw!

AlinaStarkova commented:
> If you change the wording on your PR body to something along the lines of "Fixes #issuenumber" it will close the issue once merged, btw!

huh, i added the issue number in and it didnt link the issues, even though another similar PR i did, did link. i was wondering why it didnt link this time

AlinaStarkova committed:
Update bodybag.dm

changes how it gets the location

tgstation-server commented:
Your Fix/Feature pull request delta is now above zero (0). Feel free to make Feature/Tweak/Balance PRs.
